### PR TITLE
Set custom serial number

### PR DESCRIPTION
**Pull request by okyeron**
Finally submitting something from my issue #85 from last year.

Tested with Adafruit NeoTrellis M4

usage:
```
char serialstr[32] = "mydev4676123";
USBDevice.setSerialDescriptor(serialstr);
```

and macOS reports `Serial Number:	mydev4676111` in the USB System Report for the device.

### CODE REVIEW

**Comment by TheKitty** [comment]
image
Failed CI

File: src/arduino/Adafruit_USBD_Device.cpp [hunk 1]
```diff
@@ -316,8 +321,11 @@ uint16_t const *Adafruit_USBD_Device::descriptor_string_cb(uint8_t index,
     break;
 
   case STRID_SERIAL:
-    chr_count = getSerialDescriptor(_desc_str);
-    break;
+    if (!_desc_str_arr[STRID_SERIAL]) {
+      chr_count = getSerialDescriptor(_desc_str);
```
Comment (hathach) [inline src/arduino/Adafruit_USBD_Device.cpp:325]: this code should be moved insided getSerialDescriptor()

**Review by hathach — CHANGES_REQUESTED** [review]
Thank you for your PR, `getSerialDescriptor()` must return the active serial which is the one set by application. Serial selection code must be moved inside the get function

File: src/arduino/Adafruit_USBD_Device.cpp (hunk 1 above)
Comment (okyeron) [reply src/arduino/Adafruit_USBD_Device.cpp]: Updated. Hope that's what you were looking for.

**Review by hathach — APPROVED** [review]
perfect, thank you for your PR

**Pull request merged**